internal/checker/terraform/rules/azure/azurerm_app_service.rego

```diff
@@ -10,41 +10,52 @@ package lib.terraform.CB_TFAZR_045
 
 import future.keywords.in
 
-supportedResource := [
+supportedResources := [
 	"azurerm_linux_web_app",
 	"azurerm_windows_web_app",
 	"azurerm_app_service",
 ]
 
 isvalid(block) {
 	block.Type == "resource"
-	block.Labels[_] == supportedResource[_]
+	some label in block.Labels
+	label in supportedResources
+}
+
+resource[resource] {
+	some block in pass
+	resource := concat(".", block.Labels)
+}
+
+resource[resource] {
+	some block in fail
+	resource := concat(".", block.Labels)
 }
 
 pass[resource] {
-	resource := input[_]
+	some resource in input
+	some innerBlock in resource.Blocks
 	isvalid(resource)
-	innerBlock := resource.Blocks[_]
 	innerBlock.Type == "auth_settings"
 	innerBlock.Attributes.enabled == true
 }
 
```

Comment:
add another pass block to check v2
pass[resource] {
	some resource in input
	some innerBlock in resource.Blocks
	isvalid(resource)
	innerBlock.Type == "auth_settings_v2"
	innerBlock.Attributes.auth_enabled == true
}

Reply:
thank you, this is noted, I will add a pass block for auth_settings_v2